config.yaml

```diff
@@ -12,3 +12,111 @@ options:
       Acceptable values are: "info", "debug", "warning", "error" and "critical"
     default: "info"
     type: string
+
+  host:
+    description: The hostname of the Temporal server.
+    default: ""
+    type: string
+
+  namespace:
+    description: Temporal namespace the worker should connect to.
+    default: ""
+    type: string
+
+  queue:
+    description: Temporal task queue the worker should connect to
+    default: ""
+    type: string
+
+  encryption-key:
+    description: Base64-encoded key used for data encryption.
+    default: ""
```

Comment:
This should ideally be using juju secrets if this charm is only designed to support juju 3 and up, if it needs to support juju 2.9 then it will have to live here like this.

Reply:
You're right, looking at some other charms (e.g. [postgresql-k8s](https://charmhub.io/postgresql-k8s)) it seems like the practice is to include sensitive data such as passwords as part of the charm's data bucket through peer relation. As this is not a published charm yet, I will keep it as is and refactor it in the coming PRs.

@kian99 any other opinions on this?

Reply:
I've seen the postgres-k8s charm accept secret config through actions which are then stored in the peer relation data buckets. I don't particularly like that approach because it requires manual setup after a deployment. But it is a possibility. My other comment above around whether you support juju 2.9 or 3 will also help lead your decision.

Reply:
Similar to the other Temporal charms, this one will support juju 3. Regardless of the approach, will we not require a manual step if we are not using config? (i.e. an action that creates a juju secret, until user secrets are made available).

Reply:
After a discussion with members of the Charmhub community, it seems that having these configuration secrets in config is the way to go for now until juju user secrets are implemented in juju 3.3 by the end of this cycle.

Reply:
Good enough for me.